infrastructure/ansible/dynamic.inventory/TO.py

```diff
@@ -0,0 +1,241 @@
+            flat_server_profile = "server_profile|" + server['profile']
+            flat_cachegroup = "cachegroup|" + server['cachegroup']
+            flat_server_type = "server_type|" + server['type']
+            flat_server_cdn_name = "server_cdnName|" + server['cdnName']
+            flat_server_status = "server_status|" + server['status']
+            if flat_server_profile not in out.keys():
+                out['server_profile']['children'].append(flat_server_profile)
+                out[flat_server_profile] = self.populate_server_profile_vars(
+                    traffic_ops_api,
+                    server['profileId'])
+            out[flat_server_profile]['hosts'].append(fqdn)
+            if flat_cachegroup not in out.keys():
+                out['cachegroup']['children'].append(flat_cachegroup)
+                # out[flat_cachegroup] = self.populate_cachegroups(
+                cgdata = self.populate_cachegroups(
+                    traffic_ops_api,
+                    server['cachegroupId'])
+                out[flat_cachegroup] = cgdata.cgvars
+                flat_parent_cg = cgdata.primary_parent_group_name
+                flat_second_parent_cg = cgdata.secondary_parent_group_name
+                if flat_parent_cg not in out.keys():
+                    out[flat_parent_cg] = {}
+                    out[flat_parent_cg]['children'] = []
+                if flat_second_parent_cg not in out.keys():
+                    out[flat_second_parent_cg] = {}
+                    out[flat_second_parent_cg]['children'] = []
+                out[flat_parent_cg]['children'].append(flat_cachegroup)
+                out[flat_second_parent_cg]['children'].append(flat_cachegroup)
+            out[flat_cachegroup]['hosts'].append(fqdn)
+            if flat_server_type not in out.keys():
+                out['server_type']['children'].append(flat_server_type)
+                out[flat_server_type] = {}
+                out[flat_server_type]['hosts'] = []
+            out[flat_server_type]['hosts'].append(fqdn)
+            if flat_server_cdn_name not in out.keys():
+                out['server_cdnName']['children'].append(flat_server_cdn_name)
+                out[flat_server_cdn_name] = {}
+                out[flat_server_cdn_name]['hosts'] = []
+            out[flat_server_cdn_name]['hosts'].append(fqdn)
+            if flat_server_status not in out.keys():
+                out['server_status']['children'].append(flat_server_status)
+                out[flat_server_status] = {}
+                out[flat_server_status]['hosts'] = []
+            out[flat_server_status]['hosts'].append(fqdn)
+        return out
+
+    def to_inventory(self):
+        return self.generate_inventory_list(self.to_url)
+
+
+if __name__ == "__main__":
+    PARSER = argparse.ArgumentParser(
+        description='Generate ansible inventory from TrafficOps')
+
+    PARSER.add_argument(
+        '-to',
```

Comment:
You're mixing GNU-standard long options with these single-dash long options. To avoid confusion, you should pick one - preferably the GNU-standard one.

Ideally, though, this would actually be two separate options. You're supplying two different (albeit related) pieces of information in an undocumented format. It'd be much easier to understand e.g. `-u`/`--user` and `-p`/`--password`. Especially because with `nargs='+'` you could pass `-to test` with no `$TO_PASSWORD` defined and wind up with an undefined password, which doesn't appear to be checked for anywhere.

Reply:
Addressed with commit.